File: code/ex1018.java
import java.util.*;

class Solution {
    public List<Boolean> prefixesDivBy5(int[] A) {
        List<Boolean> res = new ArrayList<>();
        
        int num = 0;
        
        for (int i = 0; i < A.length; i++) {
            num = (num * 2 + A[i]) % 5;
            if (num == 0) {
                res.add(true);
            }
            else    res.add(false);
        }
        
        return res;
    }
}

File: code/ex1018.cpp
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& A) {
        vector<bool> res;
        int cur = 0;
        for (int a : A) {
            cur = ((cur << 1) + a) % 5;
            res.push_back(cur == 0);
        }
        return res;
    }
};